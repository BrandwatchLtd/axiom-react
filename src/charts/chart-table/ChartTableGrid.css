.ax-chart-table__grid-container {
  position: relative;
}

.ax-chart-table__grid {
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  z-index: -1;
}
