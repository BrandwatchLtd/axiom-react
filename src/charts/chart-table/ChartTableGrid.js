import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { Base } from 'bw-axiom';

export default class ChartTableGrid extends Component {
  static propTypes = {
    children: PropTypes.node.isRequired,
    count: PropTypes.number,
  };

  static defaultProps = {
    count: 11,
  };

  static contextTypes = {
    labelColumnWidth: PropTypes.string.isRequired,
  };

  render() {
    const { labelColumnWidth } = this.context;
    const { children, count, ...rest } = this.props;
    const increments = Array.apply(null, { length: count }).map((_, i) => i * (100 / (count - 1)));

    return (
      <Base { ...rest } className="ax-chart-table__grid-container">
        <div
            className="ax-chart-table__grid"
            style={ { left: labelColumnWidth } }>
          <div className="ax-chart-table__grid-lines">
            { increments.map((x) =>
              <div
                  className="ax-chart-table__grid-line"
                  key={ x }
                  style={ { left: `${x}%` } } />
            ) }
          </div>
        </div>
        { children }
      </Base>
    );
  }
}
