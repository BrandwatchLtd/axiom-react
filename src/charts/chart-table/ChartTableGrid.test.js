import React from 'react';
import renderer from 'react-test-renderer';
import ChartTableGrid from './ChartTableGrid';

function getComponent(props = {}) {
  return renderer.create(
    <ChartTableGrid { ...props }  />
  );
}

describe('ChartTableGrid', () => {
  it('renders with defaultProps', () => {
    const component = getComponent();
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with count', () => {
    const component = getComponent({ count: 5 });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
