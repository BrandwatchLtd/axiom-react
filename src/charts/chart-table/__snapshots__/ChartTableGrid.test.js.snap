// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChartTableGrid renders with count 1`] = `
<div
  className="ax-chart-table__grid-container"
>
  <div
    className="ax-chart-table__grid"
    style={
      Object {
        "left": undefined,
      }
    }
  >
    <div
      className="ax-chart-table__grid-lines"
    >
      <div
        className="ax-chart-table__grid-line"
        style={
          Object {
            "left": "0%",
          }
        }
      />
      <div
        className="ax-chart-table__grid-line"
        style={
          Object {
            "left": "25%",
          }
        }
      />
      <div
        className="ax-chart-table__grid-line"
        style={
          Object {
            "left": "50%",
          }
        }
      />
      <div
        className="ax-chart-table__grid-line"
        style={
          Object {
            "left": "75%",
          }
        }
      />
      <div
        className="ax-chart-table__grid-line"
        style={
          Object {
            "left": "100%",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`ChartTableGrid renders with defaultProps 1`] = `
<div
  className="ax-chart-table__grid-container"
>
  <div
    className="ax-chart-table__grid"
    style={
      Object {
        "left": undefined,
      }
    }
  >
    <div
      className="ax-chart-table__grid-lines"
    >
      <div
        className="ax-chart-table__grid-line"
        style={
          Object {
            "left": "0%",
          }
        }
      />
      <div
        className="ax-chart-table__grid-line"
        style={
          Object {
            "left": "10%",
          }
        }
      />
      <div
        className="ax-chart-table__grid-line"
        style={
          Object {
            "left": "20%",
          }
        }
      />
      <div
        className="ax-chart-table__grid-line"
        style={
          Object {
            "left": "30%",
          }
        }
      />
      <div
        className="ax-chart-table__grid-line"
        style={
          Object {
            "left": "40%",
          }
        }
      />
      <div
        className="ax-chart-table__grid-line"
        style={
          Object {
            "left": "50%",
          }
        }
      />
      <div
        className="ax-chart-table__grid-line"
        style={
          Object {
            "left": "60%",
          }
        }
      />
      <div
        className="ax-chart-table__grid-line"
        style={
          Object {
            "left": "70%",
          }
        }
      />
      <div
        className="ax-chart-table__grid-line"
        style={
          Object {
            "left": "80%",
          }
        }
      />
      <div
        className="ax-chart-table__grid-line"
        style={
          Object {
            "left": "90%",
          }
        }
      />
      <div
        className="ax-chart-table__grid-line"
        style={
          Object {
            "left": "100%",
          }
        }
      />
    </div>
  </div>
</div>
`;
